src/libcore/slice.rs

```diff
@@ -684,13 +666,11 @@ macro_rules! iterator {
             fn next(&mut self) -> Option<$elem> {
                 // could be implemented with slices, but this avoids bounds checks
                 unsafe {
-                    ::intrinsics::assume(!self.ptr.is_null());
-                    ::intrinsics::assume(!self.end.is_null());
```

Comment:
helping llvm to not insert redundant null checks in certain loops. Ask @dotdash and @huonw